spylib/token.py

```diff
@@ -135,10 +152,10 @@ async def shoprequest(self, goodstatus, debug, endpoint, **kwargs) -> Dict[str,
                 jresp = response.json()
                 # Recalculate the rate to be sure we have the right one.
                 calllimit = response.headers['X-Shopify-Shop-Api-Call-Limit']
-                self.max_tokens = int(calllimit.split('/')[1])
+                self.rest_bucket_max = int(calllimit.split('/')[1])
                 # In Shopify the bucket is emptied after 20 seconds
                 # regardless of the bucket size.
-                self.rate = int(self.max_tokens / 20)
+                Token.rest_leak_rate = int(self.rest_bucket_max / 20)
```

Comment:
If it's a class variable then there will be one leak rate for all tokens. An app could be instead on a regular Shopify store and on a Shopify plus store, with two different leak rates.
```suggestion
                self.rest_leak_rate = int(self.rest_bucket_max / 20)
```